python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -358,6 +358,14 @@ async def process_all_batches():
             _validate_batch_output(out_batch)
             yield out_batch
 
+        # Drain the queue to yield any remaining results.
+        while not output_batch_queue.empty():
```

Comment:
just update the above condition to `while not future.done() or not output_batch_queue.empty()`?